src/ComposableCoW.sol

```diff
@@ -91,6 +113,28 @@ contract ComposableCoW is ISafeSignatureVerifier {
         }
     }
 
+    /**
+     * Authorise a single conditional order and store a value from on-chain in the cabinet
+     * @param params The parameters of the conditional order
+     * @param factory A factory from which to get a value to store in the cabinet
+     * @param data Implementation specific off-chain data
+     * @param dispatch Whether to dispatch the `ConditionalOrderCreated` event
+     */
+    function createWithContext(
+        IConditionalOrder.ConditionalOrderParams calldata params,
+        IValueFactory factory,
+        bytes calldata data,
+        bool dispatch
+    ) external {
+        create(params, dispatch);
```

Comment:
not sure if there's gas optimizations possible here, or if it's not even worth caring, but I just noted that we hash the params twice (once inside the good old `create` and the other to store in the `cabinet`